### PR TITLE
StudentParser was moved to another project

### PR DESCRIPTION
**Pull request by FirdavsIbra**

### CODE REVIEW

**Comment by FirdavsIbra** [comment]
@shurik86

File: ITechArt.Parsers/Parsers/StudentParser.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            }
+            return students.ToArray();
+        }
+
+        /// <summary>
+        /// Parse student's file from xml.
+        /// </summary>
+        public async Task<IStudent[]> XmlParseAsync(IFormFile file)
+        {
+            using var fileStream = new MemoryStream();
```
Comment (shurik86) [inline ITechArt.Parsers/Parsers/StudentParser.cs]: On our lessons we discuss about IAsyncDisposable and using. 
Can you change your code?

File: ITechArt.Parsers/Parsers/StudentParser.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+            await file.CopyToAsync(fileStream);
+
+            fileStream.Position = 0;
+
+            XmlDocument xmlDocument = new XmlDocument();
+            xmlDocument.Load(fileStream);
```
Comment (shurik86) [inline ITechArt.Parsers/Parsers/StudentParser.cs:78]: Do you try create xml parser with model and attibutes?

File: ITechArt.Parsers/Parsers/StudentParser.cs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            var nodes = xmlDocument.SelectNodes("/students/student");
+
+            IList<StudentDto> students = new List<StudentDto>(nodes.Count);
+
+            for (int node = 0; node < nodes.Count; node++)
+            {
+                StudentDto student = new()
+                {
+                    FirstName = nodes[node]["FirstName"].InnerText,
+                    LastName = nodes[node]["LastName"].InnerText,
```
Comment (shurik86) [inline ITechArt.Parsers/Parsers/StudentParser.cs]: What semantic mean nodes[node] ?

File: ITechArt.Parsers/Parsers/StudentParser.cs [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    {
+        public StudentMap()
+        {
+            Map(s => s.FirstName).Name("FirstName");
+            Map(s => s.LastName).Name("LastName");
+            Map(s => s.Email).Name("Email");
+            Map(s => s.Password).Name("Password");
+            Map(s => s.Majority).Name("Majority");
+            Map(s => s.Gender).Name("Gender");
+            Map(s => s.DateOfBirth).Name("DateOfBirth");
```
Comment (shurik86) [inline ITechArt.Parsers/Parsers/StudentParser.cs:113]: If value in cell of DateOfBirth will me 'sdfsdfdsf' - how your code will work?

**Review by shurik86 — COMMENTED** [review]
Create changes with comments, and give me answers please

File: ITechArt.Parsers/Parsers/StudentParser.cs (hunk 1 above)
Comment (FirdavsIbra) [reply ITechArt.Parsers/Parsers/StudentParser.cs]: Changed

File: ITechArt.Parsers/Parsers/StudentParser.cs (hunk 2 above)
Comment (FirdavsIbra) [reply ITechArt.Parsers/Parsers/StudentParser.cs:78]: Yes i tried to use XmlSerializer, but i have some problems, that's why it is not ready yet

File: ITechArt.Parsers/Parsers/StudentParser.cs (hunk 3 above)
Comment (FirdavsIbra) [reply ITechArt.Parsers/Parsers/StudentParser.cs]: node is the iteration of the for, yes it is unclear, so i changed it to foreach

File: ITechArt.Parsers/Parsers/StudentParser.cs (hunk 4 above)
Comment (FirdavsIbra) [reply ITechArt.Parsers/Parsers/StudentParser.cs:113]: Will be thrown NotSupportedException, cause i have helper that convert string to dateonly. And there is exception, if the value is not valid.

**Pull request merged**